prow/README.md

```diff
@@ -1,71 +1,159 @@
+   bot account. This should be done from [metal3-io-bot](https://github.com/metal3-io-bot)
+   GitHub bot account. You can follow this [link](https://github.com/settings/tokens)
+   to create the token. When generating the token, ensure to have the following scopes checked in.
+
+   - `public_repo` and `repo:status`
+   - `repo scope` for private repos
+   - `admin:org_hook` for a github org
+
+    Create a secret out of that token:
+    ```shell
+    $ kubectl create secret generic github-token --from-file=token=/path/github-token
+    ```
+
+1. Create a secret with the GCS credentials
+
+   ```shell
+   $ kubectl create secret generic gcs-credentials --from-file=service-account.json=service-account.json
+   ```
+
+1. While applying prow manifests, we are going to create an Ingress object to redirect traffic to deck and hook services.
+   We are installing [NGINX Ingress controller](https://kubernetes.github.io/ingress-nginx/deploy/) with helm, but you can
+   install your preferred one.
+
+   ```shell
+   helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+   helm install ingress-nginx/ingress-nginx --set controller.hostNetwork=true,controller.service.type="",controller.kind=DaemonSet --generate-name
+   helm repo update
+   ```
+
+1. Apply Prow manifests
+
+   ```shell
+   kubectl apply -f manifests/*
+   ```
+
+   This will create a couple of Deployments, Services, ServiceAccounts, Roles, RoleBindings and Ingress.
+
+## Secure Ingress with Cert-Manager and Let's Encrypt
+
+1. Install [cert-manager](https://github.com/jetstack/cert-manager)
+
+   ```
+   $ helm repo add jetstack https://charts.jetstack.io
+   $ helm repo update
+   $ helm install \
+   cert-manager jetstack/cert-manager \
+   --namespace cert-manager \
+   --create-namespace \
+   --version v1.3.1 \
+   --set installCRDs=true
+   ```
 
-After running tackle, it will do most of the setup but hang waiting for an
-Ingress resource to get initialized.  I killed tackle at that point.
+   Verify that you have the following pods running
+   ```shell
+   $ kubectl get pods --namespace cert-manager
 
-```
-oc delete ingress ing
-oc expose service hook --hostname=prow.apps.ci.metal3.io --path="/hook"
-oc expose service deck --hostname=prow.apps.ci.metal3.io
-```
+     NAME                                       READY   STATUS    RESTARTS   AGE
+     cert-manager-5c6866597-zw7kh               1/1     Running   0          2m
+     cert-manager-cainjector-577f6d9fd7-tr77l   1/1     Running   0          2m
+     cert-manager-webhook-787858fcdb-nlzsq      1/1     Running   0          2m
+   ```
 
-This just exposes the routes on http.  Edit each route and add the following
-yaml snippet to the spec to turn on https.
+1. Create a secret containing your Cloudflare API token
 
-```
-oc edit route hook
-oc edit route deck
-```
+   ```shell
+   $ kubectl create secret generic cloudflare-api-token-secret --from-literal "apikey=<API_KEY>" --namespace=cert-manager
```

Comment:
Would be nice to have more information why do we need `cloudflare-api-token-secret` and the role of it in the prow setup.

Reply:
Added some notes.